Create new mobile worker role automatically for new domains

esoergel opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

## Technical Summary
https://docs.google.com/document/d/1zDMHtvZAk8t0oNcuRxgzCBnyFHkGwb2m-C0o6hZMCjg/edit#
https://dimagi-dev.atlassian.net/browse/USH-2006
Should have no user-facing effects yet.  This creates the new role automatically for new domains.  After this is live, we can do a migration to back-populate it for all old domains.  There's some code clean-up in here too.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
This will undergo QA after the next phase of dev, once it's user-facing.  For now it should all be hidden from view.

### Automated test coverage

Included

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->


### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

esoergel committed:
remove unused method

esoergel committed:
Create mobile worker role for all new domains

esoergel committed:
Simplify preset roles handling

The preset_map method was only used through one helper method, which was
only called with initial roles.  I removed READ_ONLY_NO_REPORTS from
that dict (since it's not an initial role) and combined it with the
INITIAL_ROLES const, which leads to some other simplifications.

esoergel committed:
None of these things are referenced anymore

esoergel committed:
mv UserRolePresets to role_utils (models is huge)

esoergel committed:
We're not actually translating these anywhere

As mentioned in the TODO I deleted here.  Just remove uggettext_noop.

esoergel committed:
Set is_commcare_user_default flag automatically

AddisonDunn committed:
hide default mobile worker role in UI

esoergel committed:
Temporarily hide new role so we can release later

dannyroberts commented:
Is this a new default role that can be applied to `WebUser`s (and I guess to `CommCareUser`s aka mobile workers too)? If so I think the idea of calling that role "Mobile Worker" is kind of confusing, since that's also what we call, well, mobile workers.

Maybe a bit more context would make this clearer for me. Could you motivate this for me and see if that changes the way I'm looking at it?

esoergel committed:
Tell my terminal that this is a bash file

esoergel committed:
translations (sans timestamp)

Robert-Costello commented on corehq/apps/users/role_utils.py:
```diff
@@ -1,4 +1,32 @@
+from corehq.apps.users.models import UserRole, Permissions
+
+
+class UserRolePresets:
```
This seems like a much nicer/cleaner way to structure this, getting rid of `get_permissions` and `get_preset_map` etc. I'm curious if there's some standard for having it in `utils` vs `models`?

esoergel replied:
Thanks, yeah I was happy to be able to clean it up a bit.

I moved it out of models because `UserRolePresets` is only used in this `role_utils` file and in tests, so it's nice to keep it local.  Also, models files should be focused on db models, which this isn't, and the files (especially `users/models.py`) tend to get really massive and unwieldly, so it's nice to break things out of them when it is clean to do so.

dannyroberts commented:
Thanks @esoergel that helps, and it makes a lot of sense. I think my two cents might be like you said to call the role something like "Mobile Worker Default" that makes it clear that it's not the only role that can be assigned to a mobile worker, and that assigning it to a Web User doesn't "make" them a mobile worker. But really I'll defer to any informed decision that's considered it, as this is just based on my own first instinct and nothing else!

esoergel committed:
Rename role to "Mobile Worker Default"

esoergel committed:
Merge branch 'master' of github.com:dimagi/commcare-hq into es/dmw-role

esoergel committed:
translations (not mine)

I think we're in a weird space right now where not all open PRs have
had the translations test run, so they can be merged with out-of-date
translations files
